okhttp/src/main/java/okhttp3/HttpUrl.kt

```diff
@@ -1528,25 +1541,24 @@ class HttpUrl internal constructor(builder: Builder) {
     }
 
     /**
-     * Cuts `encodedQuery` up into alternating parameter names and values. This divides a query
-     * string like `subject=math&easy&problem=5-2=3` into the list `["subject", "math", "easy",
-     * null, "problem", "5-2=3"]`. Note that values may be null and may contain '=' characters.
+     * Cuts [this] up into alternating parameter names and values. This divides a query string like
+     * `subject=math&easy&problem=5-2=3` into the list `["subject", "math", "easy", null, "problem",
+     * "5-2=3"]`. Note that values may be null and may contain '=' characters.
      */
-    @JvmStatic
-    internal fun queryStringToNamesAndValues(encodedQuery: String): MutableList<String?> {
+    internal fun String.toQueryNamesAndValues(): MutableList<String?> {
```

Comment:
Does return need to stay mutable?

Reply:
It does. This is for a weird API where you provide the entire unparsed query, and then potentially build upon it.